determined-ai/determined
# perf: remove `Trialv1.latest_training` field.

## Description

- `latest_training` field is unused.
- the subquery is really expensive for big steps tables: for a 300k steps table, it adds ~800ms to the query on my machine.
- I've spent several hours trying to optimize it, and it's possible to do for a standalone query, but I couldn't find a way to do with within the gigantic `proto_get_trials_plus`.

Let's remove it, and if we need it again, maintain a FK on trial for a quick lookup similar to `best_validation_id`.

## Test Plan

1. Run a few big trials so your steps table is a few hundred k rows.
2. Check out trial metrics in the UI, and observe that `GetExperimentTrials` call takes ~ 1 second.

With this PR, it'd take <50ms.

## Commentary (optional)

metrics_spam.py
```
import logging
import random

import determined as det


def main(core_context, increment_by):
    loss = random.random()
    loss2 = random.random()
    loss3 = random.random()
    for batch in range(50000):
        steps_completed = batch + 1
        loss = loss * (1 - (1 if random.random() > 0.4 else -1) * random.random() * 0.001)
        loss2 = loss2 * (1 - (1 if random.random() > 0.05 else -1) * random.random() * 0.005)
        loss3 = loss3 * (1 - (1 if random.random() > 0.5 else -1) * random.random() * 0.1)
        core_context.train.report_training_metrics(
                steps_completed=steps_completed, metrics={"loss": loss, "loss2": loss2, "loss3": loss3}
        )

    core_context.train.report_validation_metrics(
            steps_completed=steps_completed, metrics={"loss": loss, "loss2": loss2, "loss3": loss3}
    )


if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO, format=det.LOG_FORMAT)

    with det.core.init() as core_context:
        main(core_context=core_context, increment_by=1)
```
metrics_spam.yaml
```
name: metrics_spam
entrypoint: python3 metrics_spam.py

searcher:
   name: single
   metric: x
   max_length: 1

max_restarts: 0
```

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
